Publisher/NatsPublisher.cs

```diff
@@ -58,26 +61,37 @@ public void PublishToSingleSubject()
 
             int counter = 0;
 
+            var tasks = new List<Task>();
             while (counter < StreamDetails.NUMBER_OF_TASKS)
             {
                 counter++;
-                TransportUnit dataToSend = new TransportUnit();
-                dataToSend.DatapointKey = counter.ToString();
-                Task.Run(async () =>
+                Console.WriteLine($"task {counter}");
+                tasks.Add(Task.Run(async () =>
                 {
+                    Console.WriteLine($"start");
+                    await using NatsConnection nats = new NatsConnection(new NatsOpts
```

Comment:
creating a connection per task (which is default 8 I think) to improve throughput.